src/lib.rs

```diff
@@ -1,6 +1,9 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//! Manages system resources that can be allocated to VMs and their devices.
+
 #![deny(missing_docs)]
-//! Dummy crate needs high-level documentation.
-/// Dummy public function needs documentation.
-pub fn it_works() {
-    assert!(true);
-}
+
+mod id_allocator;
+
+pub use crate::id_allocator::{Error as IdAllocatorError, IdAllocator};
```

Comment:
It may help to simplify the implementation to share a common Error definition at the crate level.